src/io/hdf/hdf_archive.cpp

```diff
@@ -142,8 +142,8 @@ bool hdf_archive::create(const std::filesystem::path& fname, unsigned flags)
   possible_filename_ = fname;
   if (Mode[NOIO])
     return true;
-  if (!(Mode[IS_PARALLEL] || Mode[IS_MASTER]))
-    throw std::runtime_error("Only create file in parallel or by master but not every rank!");
```

Comment:
Why are these lines offensive? Please explain your use case.

Reply:
I feel the caller side was using hdf_archive incorrectly.

Reply:
This is what blocks Traces from functioning in a parallel run.  Traces creates one H5 file per MPI process.  This does not create significant runtime overhead for QMCPACK vs having traces off (measured at about a 5% performance impact on large scale Titan runs).

Reply:
See Sec. IV A in https://doi.org/10.1109/CSE.2013.24

Reply:
hdf_archive should support one file file per rank when it is not aware of MPI. So the issue is incorrect open mode of the file not the hdf_archive itself. Please provide a link where open() is called from the trace code.

Reply:
This works.  I've restored the guards in hdf_archive now and the update for the fix is now confined to TraceManager.h